src/lcfiplus.cc

```diff
@@ -1270,6 +1270,51 @@ vector<const Track*> Jet::getAllTracks(bool withoutV0) const {
   return ret;
 }
 
+vector<const Track*> Jet::getAllTracksSorted(bool withoutV0) const {
+  vector<const Track*> tracks_unsorted = getAllTracks(withoutV0);
+
+  // order particles by energy
+  vector<std::pair<float, int> > order_tr;
+  order_tr.resize(tracks_unsorted.size());
+  for(size_t i=0; i<tracks_unsorted.size(); ++i){
+    order_tr[i] = std::pair<float, int>(tracks_unsorted[i]->E(), i);
+  }
+  std::sort(order_tr.begin(),order_tr.end(),[](std::pair<float,int>a,std::pair<float,int> b){
+    return a.first > b.first;
+  });
+
+  vector<const Track*> tracks;
+  tracks.resize(tracks_unsorted.size());
+
+  for (size_t i=0; i<tracks_unsorted.size(); ++i) {
+    tracks[i] = tracks_unsorted[order_tr[i].second];
+  }
+  return tracks;
```

Comment:
```suggestion
  auto tracks_unsorted = getAllTracks(withoutV0);
  std::ranges::sort(tracks_unsorted, {}, &Track::E);
```
Since we are building with c++20 in any case, we can also use [`std::ranges::sort`](https://en.cppreference.com/w/cpp/algorithm/ranges/sort.html) and not create an intermediate vector just for soting. Similar for the neutrals below.